[TSP Migration]--datadog

welovej opened the pull request:
This PR migrates your latest version (identified by the tag in your readme.md) of swagger to TypeSpec. We already tried our best to make sure the TypeSpec represents same as previous swagger. Since we lack the business knowledge, please validate this PR again to make sure it's functional equivalent as before. The local validation step is at [Getting started | TypeSpec Azure](https://azure.github.io/typespec-azure/docs/migrate-swagger/01-get-started/#review-and-adjust-the-typespec)

Besides, TypeSpec encourages to follow [ARM guidelines](https://aka.ms/rpguidelines). Therefore, some representations in your previous swagger will be fixed to follow these guidelines. When you see differences in your local validation, please keep this note in mind. This is the list you need to pay attention:
- Common type version is lower than 3.

Please reach out to [TypeSpec Discussions Channel](https://teams.microsoft.com/l/channel/19%3a906c1efbbec54dc8949ac736633e6bdf%40thread.skype/TypeSpec%2520Discussion%2520%25F0%259F%2590%25AE?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) if there is any help needed.

welovej committed:
newly converted

welovej committed:
update

welovej committed:
Update models.tsp

welovej committed:
update

welovej committed:
Update tspconfig.yaml

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into TspMig-datadog

weidongxu-microsoft commented on specification/datadog/Datadog.Management/main.tsp:
```diff
@@ -0,0 +1,76 @@
+using TypeSpec.Versioning;
+using Azure.ResourceManager.Private;
+
+/**
+ * // FIXME: (missing-service-description) Add service description
+ */
```
Can we delete this?

welovej replied:
sure

Commit:
update

welovej committed:
update

welovej committed:
update

JiaqiZhang-Dev commented on specification/datadog/Datadog.Management/MonitoredSubscriptionProperties.tsp:
```diff
@@ -0,0 +1,81 @@
+    MonitoredSubscriptionProperties,
+    Response = ArmResourceUpdatedResponse<MonitoredSubscriptionProperties> | ArmResourceCreatedResponse<
+      MonitoredSubscriptionProperties,
+      ArmLroLocationHeader & Azure.Core.Foundations.RetryAfterHeader
```
Should here need to specify `FinalResult`?

Commit:
update

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into TspMig-datadog

melina5656 committed:
Fix optional body issues

melina5656 committed:
Merge remote-tracking branch 'upstream/main' into TspMig-datadog

melina5656 committed:
Delete conditionalClientFlatten and run prettier check

welovej committed:
Merge remote-tracking branch 'upstream/main' into TspMig-datadog

welovej committed:
FIXME

mcgallan committed:
Merge remote-tracking branch 'upstream/main' into pr/34281

mcgallan committed:
update

mcgallan committed:
update

mcgallan committed:
update

tadelesh commented on specification/datadog/Datadog.Management/DatadogMonitorResource.tsp:
```diff
@@ -0,0 +1,193 @@
+    DatadogMonitorResource,
+    PatchModel = DatadogMonitorResourceUpdateParameters,
+    Response = ArmResponse<DatadogMonitorResource> | (ArmResourceCreatedResponse<DatadogMonitorResource> &
+      ArmLroLocationHeader),
```
Missing final result.

XiaofeiCao commented:
/azp run SDK Validation - Java

v-jiaodi committed:
Merge branch 'main' into TspMig-datadog

v-jiaodi committed:
Merge branch 'main' into TspMig-datadog

melina5656 committed:
Remove Bodyboot Decorator

v-jiaodi committed:
resolve breaking

welovej committed:
Update ArmLroLocationHeader type in monitor resource

Changed the Response type to use ArmLroLocationHeader with FinalResult set to DatadogMonitorResource for improved type specificity in DatadogMonitorResources interface.

v-jiaodi commented on specification/datadog/resource-manager/Microsoft.Datadog/stable/2023-10-20/datadog.json:
```diff
@@ -1777,875 +1892,1052 @@
         }
-      }
+      },
+      "required": [
+        "subscriptionId"
```
`subscriptionId` should be optional for `MonitoredSubscription`

welovej committed:
fix for subscriptionId to be  optional

johanste commented on specification/datadog/Datadog.Management/DatadogMonitorResource.tsp:
```diff
@@ -0,0 +1,192 @@
+  ...ResourceNameParameter<
+    Resource = DatadogMonitorResource,
+    KeyName = "monitorName",
+    SegmentName = "monitors",
```
Remove since this is the default value.

johanste commented on specification/datadog/Datadog.Management/DatadogMonitorResource.tsp:
```diff
@@ -0,0 +1,192 @@
+         * Monitor resource name
+         */
+        @path
+        @segment("monitors")
```
This should be the default segment for a `Monitor` resource, no?

pshao25 replied:
It is expected to have this segment part in a legacy operation

welovej committed:
Update tspconfig.yaml

welovej committed:
remove private things

welovej committed:
Update main.tsp

welovej committed:
Update tspconfig.yaml

welovej committed:
fix for as description and migration language scop decorator to client.tsp

System.ArgumentException: 'An item with the same key has already been added. Key: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}'

welovej committed:
Update client.tsp

welovej committed:
fix for typespec migration validation

Comparing finished.
| Type | Level | Message |
| ---- | ----- | ------- |
| finalresult | error | The finalresult of operation "Monitors_Create" changed: "DatadogMonitorResource" -> undefined |
| finalresult | error | The finalresult of operation "Monitors_Update" changed: "DatadogMonitorResource" -> undefined |
| finalresult | error | The finalresult of operation "SingleSignOnConfigurations_CreateOrUpdate" changed: "DatadogSingleSignOnResource" -> undefined |
| finalresult | error | The finalresult of operation "MonitoredSubscriptions_CreateorUpdate" changed: "MonitoredSubscriptionProperties" -> undefined |
| finalresult | error | The finalresult of operation "MonitoredSubscriptions_Update" changed: "MonitoredSubscriptionProperties" -> undefined |

welovej committed:
Revert "fix for typespec migration validation"

This reverts commit 80be29c216cefde5d8c8a7f243c0945e8c73c326.

welovej committed:
update

welovej committed:
fix: for @pagedItems

welovej committed:
Update DatadogMonitorResource.tsp

weidongxu-microsoft committed:
compability for java

welovej committed:
Update main.tsp
